### ISSUE TITLE
plop command should return non-0 return state when generation failed

### ISSUE THREAD

**Issue by Pike** [plopjs/plop#299]
I was just running a plop wrapper in integration tests, and noticed that the generation of the project failed, but the command succeeded.
Then I looked at plop proper, and I don't see that returning an error code back to the shell either? I only found those for internal errors, but not for failed actions?

**Comment by crutchcorn** [comment]
Hmm... I wonder if this was a change in `ora` when we updated to v3. Were you running integration tests prior to the v3 release?

I _think_ this should be easy enough. We can do something like this:

```
---
 src/plop.js | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/plop.js b/src/plop.js
index 8deeec2..b1ce456 100755
--- a/src/plop.js
+++ b/src/plop.js
@@ -121,6 +121,7 @@ async function run(env, _, passArgsBeforeDashes) {
 // everybody to the plop!
 //
 function doThePlop(generator, bypassArr) {
+  let failedActions = false;
   generator
     .runPrompts(bypassArr)
     .then(async (answers) => {
@@ -157,11 +158,15 @@ function doThePlop(generator, bypassArr) {
         }
         progressSpinner.fail(line);
         progressSpinner.start();
+        failedActions = true;
       };
       progressSpinner.start();
       return generator
         .runActions(answers, { onSuccess, onFailure, onComment })
-        .then(() => progressSpinner.stop());
+        .then(() => {
+          progressSpinner.stop();
+          if (failedActions) process.exit(1);
+        });
     })
     .catch(function (err) {
       console.error(chalk.red("[ERROR]"), err.message);
-- 
```

Would you be comfortable perhaps testing to see if this works in your environment? If so, we can add E2E tests and publish changes.

Admittedly, I might not be able to get to this properly for a few days, but PRs are welcomed before then and I can help answer any questions you might have about our E2E test setup

**Comment by crutchcorn** [comment]
This has been fixed as-of 3.0.2
